crates/payload/primitives/src/payload.rs

```diff
@@ -2,32 +2,78 @@ use crate::{MessageValidationKind, PayloadAttributes};
 use alloc::vec::Vec;
 use alloy_eips::eip4895::Withdrawal;
 use alloy_primitives::B256;
-use alloy_rpc_types_engine::ExecutionPayload;
+use alloy_rpc_types_engine::ExecutionData;
+use core::fmt::Debug;
+use serde::{de::DeserializeOwned, Serialize};
 
-/// Either an [`ExecutionPayload`] or a type that implements the [`PayloadAttributes`] trait.
+/// An execution payload.
+pub trait ExecutionPayload:
+    Serialize + DeserializeOwned + Debug + Clone + Send + Sync + 'static
+{
+    /// Returns the parent hash of the block.
+    fn parent_hash(&self) -> B256;
+
+    /// Returns the hash of the block.
+    fn block_hash(&self) -> B256;
+
+    /// Returns the number of the block.
+    fn block_number(&self) -> u64;
+
+    /// Returns the withdrawals for the payload, if it exists.
+    fn withdrawals(&self) -> Option<&Vec<Withdrawal>>;
+
+    /// Return the parent beacon block root for the payload, if it exists.
+    fn parent_beacon_block_root(&self) -> Option<B256>;
+
+    /// Returns the timestamp to be used in the payload.
+    fn timestamp(&self) -> u64;
+}
```

Comment:
needs implementation for `OpExecutionPayload` too, think @mattsse wants an `op` feature to be added for this reason for now to feature gate the impl of this trait under the impl for `ExecutionData`

Reply:
can do that next